woocommerce/woocommerce-android
# IPP Feedback Banner - final PR

### IPP Feedback Banner - Project's final PR

This is a complete feature PR consisting of the following partial PRs, reviewed and merged to the this feature branch:
1. https://github.com/woocommerce/woocommerce-android/pull/8159
2. https://github.com/woocommerce/woocommerce-android/pull/8165
3. https://github.com/woocommerce/woocommerce-android/pull/8180
4. https://github.com/woocommerce/woocommerce-android/pull/8205
5. https://github.com/woocommerce/woocommerce-android/pull/8225
6. https://github.com/woocommerce/woocommerce-android/pull/8230

Closes: https://github.com/woocommerce/woocommerce-android/issues/8116


### Description
The goal of the project is to collect feedback from IPP users. We target three groups of merchants and show dedicated banners and feedback surveys.

- Specs for the banner title and message texts: p1674116096400209/1674112248.357019-slack-C0354HSNUJH

- More details in the project thread: pdfdoF-20T-p2

### Testing instructions

The banner should be visible on the orders page. The condition required to see the banner is to have COD enabled. 
If the user has ever completed IPP transactions banner type 2. or 3. (more than 10 transactions in the last 30 days) is shown. If they haven't made any IPP transactions banner 1. is shown.

### Images
The banner is placed on the Orders screen. Three banner types are shown in the table below.

| 1. IPP newbie users  | 2. IPP beginners | 3. IPP active users |
| ------------- | ------------- | ------------- | 
| ![Screenshot_20230123-113541](https://user-images.githubusercontent.com/4527432/214398300-5eba97a6-2d05-4383-8726-c17c592b59ff.png)  | ![Screenshot_20230123-113713](https://user-images.githubusercontent.com/4527432/214398338-c26c8eda-4ffb-4226-91bd-783a5a77d0da.png)  | ![Screenshot_20230123-113834](https://user-images.githubusercontent.com/4527432/214398361-83feda1b-cb76-4ebe-8477-188d8ca00dd0.png) |
